clueless/state.py

```diff
@@ -252,16 +252,22 @@ async def add_user(self, player: Player):
 
     async def accuse(self, player: Character, accuse: Accuse):
         accusation = accuse.into_accusation(player)
-        await self.broadcast(accusation, skip=player)
-        if accusation.suspect == self.crime_character and accusation.room == self.crime_room and accusation.weapon == self.crime_weapon:
-            await self.broadcast(Winner(player))
+
+        # validate accusation objects are within dictionary of available objects
+        if 0 <= accusation.suspect <= 5 and 0 <= accusation.weapon <= 5 and 0 <= accusation.room <= 9:
```

Comment:
`suspect`, `weapon`, and `room` should all be valid instances of their classes by this point. The `accusation` object is an instance of `Accuse` created from `deserialize_message`, and these values are actually `Character`, `Weapon` and `Room` instances, not just ints (although they can be treated as ints as you have done here, since they're sub-classes of `IntEnum` which allows you to transparently "convert" from an instance of the enum to an int). If there is a number for one of these values that isn't valid, it will be caught in `deserialize_message` and the error message will be send [here](https://github.com/JHU-601/flask-production/pull/24/files?diff=split&w=1#diff-a970d717e966e16889f252f979f4a90eb15a45e8037ab2d64d6e753a0be047faR144).

So the validation that needs to happen here would just be that it is the player's turn.

So this method should probably be changed to take `player: Player` instead of `player: Character`, and then this check would be
```
if self.current_player == player:
  ...
else:
  await player.send_message(Status error message telling them ITS NOT YOUR TURN CHILL OUT)
```